usagov/usagov-2021
# USAGOV-1990 - Implement GSA Auth Single Sign On (replacing SecureAuth)

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1990

## Description


Add script which applies cf space specific config values for SAML Authentication module.
Included config sets are DR, DEV and the original, environment-agnostic SecureAuth.

(Note that we have not received the metadata for Stage and Prod yet).

The bootstrap process will use the space name from VCAP to apply the correct config set, using drush cset.

This is a lightweight process that can be used until we choose to implement config_split, if we decide to go that route.

It works well in DR, and we have the config set for dev, so it will be tested there, at first bootstrap, once merged.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

After deployment, attempt to login at /usr/login
Doing so should invoke the GSAAuth SSO process, and you should be allowed to log into the CMS (provided you have set up your GSAAuth ENT account).

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
